head/src/main/kotlin/io/qalipsis/core/head/jdbc/repository/FactoryRepository.kt

```diff
@@ -91,6 +92,31 @@ internal interface FactoryRepository : CoroutineCrudRepository<FactoryEntity, Lo
     @Join(value = "tags", type = Join.Type.LEFT_FETCH)
     suspend fun getAvailableFactoriesForScenarios(tenant: String, names: Collection<String>): List<FactoryEntity>
 
+    /**
+     * Checks factories' health in a [tenant]. In factory_state table,
+     * we only select the row corresponding to the more recent health_timestamp.
+     *
+     * @param tenant the tenant's reference
+     * @param factoryNodeIds the collection of factories to check
+     */
+    @Query(
+        """SELECT factory_max_timestamp."node_id", 
+                factory_state."health_timestamp" as "timestamp", 
+                factory_state."state" 
```

Comment:
When the latest health for a factory is "HEALTHY" but its health_timestamp is older than 2 minutes, it should be considered as UNHEALTHY. 
Please create a proper test for it.